src/main/java/games/strategy/triplea/ui/TripleAFrame.java

```diff
@@ -1048,7 +1048,7 @@ public void run() {
         for (final Entry<Territory, Collection<Unit>> entry : possibleUnitsToAttack.entrySet()) {
           final List<Unit> units = new ArrayList<>(entry.getValue());
           Collections.sort(units,
-              new UnitBattleComparator(false, TuvUtils.getCostsForTuvForAllPlayersMergedAndAveraged(data),
+              new UnitBattleComparator(false, TuvUtils.getCostsForTuv(units.get(0).getOwner(), data),
```

Comment:
Is there a possibility that `units` can be empty here?

Reply:
Nope. There is only 1 call path to this and it only inserts into the map if the collection of units isn't empty.